core-rust/state-manager/src/state_manager.rs

```diff
@@ -333,6 +362,22 @@ where
                 break;
             }
 
+            // Don't process any additional transactions if the vertex has been filled
+            // (most probable by transaction count).
+            if vertex_limits_tracker.is_full(&self.vertex_limits_config) {
+                break;
+            }
+
+            let transaction_size = raw_user_transaction.as_slice().len();
+
+            // Skip validating and executing this transaction if it doesn't fit it in the vertex.
+            if vertex_limits_tracker
+                .check_pre_execution(&self.vertex_limits_config, transaction_size)
```

Comment:
it feels a bit weird that `.is_full()` and `.check_pre_execution()` co-exist;
even weirder that they are called one after another;
and even weirder that they both check `total_transactions_count` (effective one of those is 0xDEADC0DE, isn't it?)

AFAIU, conceptually we want to distinguish between 2 cases:
A. "some limit was exceeded by this transaction, but it's worth checking another one!"
B. "some limit was exceeded by this transaction and will definitely be exceeded by any other"
(that B. is a bit artificial, because effectively only the tx count can cause this, right?)

anyways, maybe it's possible to indicate this in the result returned from `check_pre_execution()`?
(and drop `is_full()`)